appdatabase/migrations/sql/1691258252_add_scope_to_pending.up.sql

```diff
@@ -0,0 +1,20 @@
+CREATE TABLE pending_tx_status (
```

Comment:
Same question... Doesn't sound like one would normally use a SQL table for this

Reply:
I was searching for a way to document the enum values without storing the name of the enum values. It looked to me like this way might bring some advantages:

- keep only enum values (index) instead of duplicate string names
- you can still query and visualize the content in a human-readable way
```sql
SELECT SUBSTR(HEX(from_address), 0, 5), pending_tx_status.name, auto_delete
FROM pending_transactions
JOIN pending_tx_status ON pending_transactions.status = pending_tx_status.id;
```
- because this column is expected to keep an enum that is found in status-go, we can have a test to validate that the DB is consistent with the code expectations

Maybe you find this approach unnatural. Let me know what you think. For sure, this experiment can be dropped and keep the strings. The space save for `pending_transactions` is negligible, and the changes of changing enum values are low and not very error-prone.

Reply:
Thanks for the great feedback @IvanBelyakoff, I'm glad to see your perspective on this experiment and will fall back to using enum names.

> ... Still it feels to me like an overkill. You keep the declaration of the constants in migration files, which I believe even IDE can't reference to and not an obvious place for devs to find them. And I believe it is confusing for many devs raising more questions than answers.

I saw it as if it is in the migration files then it is documented. Because migrations are run it is in the DB alos. Because we have a test to link the enum consistency in the code with the DB then it is also checked and in sync with the code enumerations.

> Moreover the need to upgrade the database to change status enums does not look like a good idea as this involves DB upgrade for all users that we will have and DB upgrade is a lot more risky and troublesome thing that depends on environment than code upgrade (speaking about constants, this does not depend on any environment and a way more safe thing that does not happen in runtime, but at build time).

I see the free-roaming strings even worse in case of needing to extend the enum. If we need to change the enum (add/remove elements) we still have to migrate the data to keep the DB normalize, just that with strings there is nothing to rise a flag that the required migration was not done.

> So thinking it over once again, I am quite strongly against this peace of code

You were heard, and I will remove the reference from the DB